test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java

```diff
@@ -148,4 +159,199 @@ public static String csym(Class<?> c) {
+            Function<Class<?>, Z> resFunc, Consumer<? super C> builder) {
+
+        IsolatedMethodBuilder isolatedMethodBuilder = new IsolatedMethodBuilder(className, lookup);
+        isolatedMethodBuilder
+                .withSuperclass(Object.class)
+                .withMajorVersion(60)
+                .withMinorVersion(0)
+                .withFlags(Flag.ACC_PUBLIC)
+                .withMethod(methodName, type, M ->
+                        M.withFlags(Flag.ACC_STATIC, Flag.ACC_PUBLIC)
+                                .withCode(builderFunc, builder));
+
+        try {
+            byte[] byteArray = isolatedMethodBuilder.build();
+            Class<?> clazz = lookup.defineClass(byteArray);
+            return resFunc.apply(clazz);
+        } catch (Throwable e) {
+             throw new IllegalStateException(e);
+        }
+    }
+
+    public static class IsolatedMethodBuilder extends ClassBuilder<Class<?>, String, IsolatedMethodBuilder> {
+
+        private static final Class<?> THIS_CLASS = new Object() { }.getClass();
+
+        public IsolatedMethodBuilder(String clazz, MethodHandles.Lookup lookup) {
+            super(new IsolatedMethodPoolHelper(clazz),
+                  new IsolatedMethodTypeHelper(lookup));
+            withThisClass(THIS_CLASS);
+        }
+
+        public Class<?> thisClass() {
+            return THIS_CLASS;
+        }
+
+        static String classToInternalName(Class<?> c) {
+            return c.getName().replace('.', '/');
+        }
+
+        static class IsolatedMethodTypeHelper implements TypeHelper<Class<?>, String> {
+
+            BasicTypeHelper basicTypeHelper = new BasicTypeHelper();
+            MethodHandles.Lookup lookup;
+
+            IsolatedMethodTypeHelper(MethodHandles.Lookup lookup) {
+                this.lookup = lookup;
+            }
+
+            @Override
+            public String elemtype(String s) {
+                return basicTypeHelper.elemtype(s);
+            }
+
+            @Override
+            public String arrayOf(String s) {
+                return basicTypeHelper.arrayOf(s);
+            }
+
+            @Override
+            public Iterator<String> parameterTypes(String s) {
+                return basicTypeHelper.parameterTypes(s);
+            }
+
+            @Override
+            public String fromTag(TypeTag tag) {
+                return basicTypeHelper.fromTag(tag);
+            }
+
+            @Override
+            public String returnType(String s) {
+                return basicTypeHelper.returnType(s);
+            }
+
+            @Override
+            public String type(Class<?> aClass) {
+                if (aClass.isArray()) {
+                    return classToInternalName(aClass);
+                } else {
+                    return (aClass.isInlineClass() ? "Q" : "L") + classToInternalName(aClass) + ";";
+                }
+            }
+
+            @Override
+            public boolean isValue(String desc) {
```

Comment:
I think we should rename `isValue` to `isInlineClass` and also other new `xxxValue` methods

Reply:
Comments applied